Tests/Mailer/SymfonyAuthCodeMailerTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Scheb\TwoFactorBundle\Tests\Mailer;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use Scheb\TwoFactorBundle\Mailer\SymfonyAuthCodeMailer;
+use Scheb\TwoFactorBundle\Model\Email\TwoFactorInterface;
+use Scheb\TwoFactorBundle\Tests\TestCase;
+use Symfony\Component\Mailer\Mailer;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Mime\Email;
+
+class SymfonyAuthCodeMailerTest extends TestCase
+{
+    /**
+     * @var MockObject|TransportInterface
+     */
+    private $symfonyTransportInterface;
+
+    /**
+     * @var SymfonyAuthCodeMailer
+     */
+    private $mailer;
+
+    protected function setUp(): void
+    {
+        if (\interface_exists(TransportInterface::class) === false) {
+            $this->markTestSkipped("Symfony mailer not installed");
+
+            return;
+        }
+        $this->symfonyTransportInterface = $this->createMock(TransportInterface::class);
+        $this->mailer = new SymfonyAuthCodeMailer(new Mailer($this->symfonyTransportInterface), 'sender@example.com', 'Sender Name');
```

Comment:
Here's where you should use a mock of `MailerInterface` instead. And I guess we can also use in `\interface_exists(TransportInterface::class)` so we don't need to rely on another class to detect if the mailer component is present.